### PR TITLE
[bitnami/kafka] Use the container port instead of the service port in kafka metrics exporter

### LINKED ISSUE #18284: kafka-exporter's generated command uses the wrong port to talk to the kafka container.
### Name and Version

bitnami/kafka:24.0.3

### What architecture are you using?

amd64

### What steps will reproduce the bug?

Turn on metrics:
`
metrics:
  kafka:
    enabled: true
`
The generated command for the metrics deployment's port points to $servicePort and 
`{{- $servicePort := int .Values.service.ports.client -}}` 
but it should be the container port (not the service port) since the headless form of the container FQDN is being used
`{{- $servicePort := int .Values.listeners.client.containerPort -}}`

### Are you using any custom parameters or values?

_No response_

### What is the expected behavior?

When using the headless form of the container FQDN, the container port should be used.

### What do you see instead?

The headless service port is being used which does not resolved if tagged to the headless form of the container FQDN.

### Additional information

_No response_

### PR DESCRIPTION
**Pull request by jadolg**
### Description of the change

In the Kafka chart, when metrics are enabled, the exporter needs to point to all pods, not to the service.
This change makes it so the port used to reach the pods is the actual pod port and not the service port.

### Benefits

Makes the exporter start working correctly again

### Applicable issues

- fixes #18284 

### Checklist

- [x] Chart version bumped in `Chart.yaml` according to [semver](http://semver.org/). This is *not necessary* when the changes only affect README.md files.
- [x] Title of the pull request follows this pattern [bitnami/<name_of_the_chart>] Descriptive title
- [x] All commits signed off and in agreement of [Developer Certificate of Origin (DCO)](https://github.com/bitnami/charts/blob/main/CONTRIBUTING.md#sign-your-work)

### CODE REVIEW

**Review by jotamartos — DISMISSED** [review]
Thank you for your contribution. However, the branch has conflicts that needs to be resolved. You can learn more about that below.

**Comment by migruiz4** [comment]
Hi @jadolg,

Could you please update your branch again? A new version was released but GitHub won't let me solve conflicts in the UI.

**Comment by jadolg** [comment]
> Hi @jadolg,
> 
> Could you please update your branch again? A new version was released but GitHub won't let me solve conflicts in the UI.

Done

File: bitnami/kafka/Chart.yaml [hunk 1]
```diff
@@ -42,4 +42,4 @@ maintainers:
 name: kafka
 sources:
 - https://github.com/bitnami/charts/tree/main/bitnami/kafka
-version: 25.1.1
+version: 24.0.16
```
Comment (jotamartos) [inline bitnami/kafka/Chart.yaml:45]:
```suggestion
version: 25.1.2
```

**Review by jotamartos — CHANGES_REQUESTED** [review]
Please bump the Chart version accordingly.

**Review by jotamartos — APPROVED** [review]

**Comment by PDVJAM** [comment]
25.1.2 chart still has service ports instead of container ones.

**Comment by jotamartos** [comment]
Hi @PDVJAM,

Could you please provide more information? I just downloaded the latest bitnami/kafka chart and saw that these changes are included in the chart's templates. 

Thanks

**Comment by PDVJAM** [comment]
@jotamartos Hey. Yes, now all ok. But the 25.1.2 was still not updated, but according to the merge - it should have.

**Pull request merged**